content/api/configuration.md

```diff
@@ -1,10 +1,1407 @@
+
+`string`
+
+Configure how the library will be exposed. Any one of the following options can be used:
+
+`libraryTarget: "var"` - Expose it as a variable (i.e. `var MyLibrary = ...`)
+
+`libraryTarget: "this"` - Expose it as a property of `this` (i.e. `this.MyLibrary = ...`)
+
+`libraryTarget: "commonjs"` - Expose it using the `exports` object (i.e. `exports["MyLibrary"] = ...`)
+
+`libraryTarget: "commonjs2"` - Expose it using the `module.exports` object
+
+`libraryTarget: "amd"` - Expose it using [Asynchronous Module Defintion]() (AMD)
+
+`libraryTarget: "umd"` - Expose it using [Universal Module Definition]() (UMD)
+
+---
+
+### `output.umdNamedDefine`
+
+`boolean`
+
+When using `libraryTarget: "umd"`, setting:
+
+```js
+umdNamedDefine: true
+```
+
+will name the AMD module of the UMD build.
+
+---
+
+### `output.sourcePrefix`
+
+`string`
+
+Change the prefix for each line in the output bundles. The default is tabs:
+
+```js
+sourcePrefix: "\t"
+```
+
+T> This option can be helpful in [fixing issues with multiline strings]().
+
+---
+
+### `output.crossOriginLoading`
+
+`boolean` `string`
+
+Enable [cross-origin]() loading of [chunks](). The following values are accepted...
+
+`crossOriginLoading: false` - Disables cross-origin loading (default)
+
+`crossOriginLoading: "anonymous"` - Enable cross-origin loading **without credentials**
+
+`crossOriginLoading: "use-credentials"` - Enable cross-origin loading **with credentials**
+
+---
+
+### `module`
+
+`object`
+
+These options determine how the [different types of modules](/concepts/everything-is-a-module) within a project will be treated.
+
+---
+
+### Loader Objects
+
+Loader objects are used in a few places throughout the configuration. They identify groups of modules using regular expressions. [Loaders](/concepts/loaders) can then be used, and chained together, to process, transform, or manipulate that group of modules. Loader objects can contain the following properties:
+
+`test: /\.js/` - Identify one or more file extensions using a [regex](), string, or function
+
+`include: /\/src/` - Include modules using a regex, string, or function
+
+`exclude: /node_modules/` - Exclude modules using a regex, string, or function
+
+`loader: "babel!eslint"` - A ! delimited string of loaders to use on these modules
+
+`loaders: [ "babel", "eslint" ]` - An array of loaders to use on these modules
+
+W> Note that loaders are always read from **right to left** whether passed via a delimited string or an array. In the example above, the [eslint-loader]() will lint, and possibly fix syntax in, the JavaScript modules and then hand them off to the [babel-loader]() for transpiling.
+
+---
+
+### `module.preLoaders`
+
+`array`
+
+An array of [loader objects]() to be used as the first step in the loading process. In the example above, linting could be broken out into a *preLoader*:
+
+```js
+module: {
+  preLoaders: [
+    { test: /\.js/, exclude: /node_modules/, loader: 'eslint' }
```

Comment:
We probably should use `-loader` suffix for each `loader` definition for clarity. See https://github.com/webpack/webpack/issues/2986 .